main.py

```diff
@@ -0,0 +1,89 @@
+import json
+import os
+import requests
+import re
+
+
+owner, repo_name = os.environ["OWNER_AND_REPO"].split("/")
+
+url = "https://api.github.com/graphql"
+query = """
+query getDiscussions {
+    repository(name: "$repo_name", owner: "$owner") {
+        discussions(first: 100) {
+            nodes {
+                title
+                body
+                author {
+                    login
+                }
+            }
+            totalCount
+        }
+    }
+}
+"""
+
+variables = {
+    "repo_name": repo_name,
+    "owner": owner,
+}
+
+def apply_variables(query, variables):
+    for name, value in variables.items():
+        query = query.replace("$"+name, str(value))
+
+    return query
+
+def parse_discussions(response):
+    # the multiline_re matches any character including newlines
+    multiline_re = r"[\s\S]*"
+    # regexes for parsing the entries in discussion body
+    # multiline_re is needed to match even if there are forbidden new lines
+    description_re = r"\*\*Description\*\*: (.*)" + multiline_re
+    url_re = r"\*\*URLs\*\*: (.*)\n" + multiline_re
+    types_re = r"\*\*Types\*\*: (.*)\n" + multiline_re
+    platforms_re = r"\*\*Platforms\*\*: (.*)\n" + multiline_re
+    # additional lines are considered only in comments section
+    comments_re = r"\*\*Comments\*\*: " + r"(" + multiline_re + r")"
+
+    body_regex = re.compile(description_re + url_re +
+                            types_re + platforms_re + comments_re + r"$")
+
+    print("Printing fetched and parsed discussions:")
+    for discussion in json.loads(response.text)["data"]["repository"]["discussions"]["nodes"]:
+        print("-----------------------------------------------------------------------------------------------------------")
+        app_name = discussion["title"]
+        m = body_regex.match(discussion["body"])
+
+        try:
+            description = m.group(1)
+            urls = m.group(2)
+            types = m.group(3)
+            platforms = m.group(4)
+            comments = m.group(5)
+        except (AttributeError):
+            print("Error during body parsing for " + app_name)
+            continue
+
+        print("App_name: " + app_name)
+        print("Description: " + description)
+        print("Urls: " + urls)
+        print("Types: " + types)
+        print("Platforms: " + platforms)
+        print("Comments: " + comments)
```

Comment:
Create a f string and then print it out. One print statement instead of several.

```suggestion

      print(f"App Name: {app_name}\n"
      f"Description: {description}\n"
      f"URLs: {urls}\n"
      f"Types: {types}\n"
      f"Platforms: {platforms}\n"
      f"Comments: {comments}")        
```

Could be assigned to variable. This is even a better option.